katsdpimager/preprocess.cpp

```diff
@@ -432,78 +435,79 @@ void visibility_collector<P>::add_impl2(
         float uv_scale = 1.0f / conf.cell_size;
         float w_scale = (conf.w_slices - 0.5f) * conf.w_planes / conf.max_w;
         int max_slice_plane = conf.w_slices * conf.w_planes - 1; // TODO: check for overflow? precompute?
-        for (std::size_t i = 0; i < N; i++)
+        for (std::size_t i0 = 0; i0 < N; i0 += buffer_capacity)
         {
-            if (weights.col(i).cwiseEqual(0.0f).any())
+            std::size_t i1 = std::min(N, i0 + buffer_capacity);
+#pragma omp parallel for schedule(static)
+            for (std::size_t i = i0; i < i1; i++)
```

Comment:
So is this processing `buffer_capacity` consecutive visibilities in parallel?

Reply:
It's processing [i0, i1) in parallel - so up to `buffer_capacity`, but less for the last batch.